internal/service/advancedclustertpf/resource.go

```diff
@@ -313,10 +336,14 @@ func (r *rs) updateLegacyReplicationSpecs(ctx context.Context, state, plan *TFMo
 		}
 	}
 	numShardsPlan := numShardsMap(ctx, plan.ReplicationSpecs, diags)
+	legacyIDs := externalIDToLegacyID(ctx, state.ReplicationSpecs, diags)
```

Comment:
q: In the case of update for legacy replication specs, are we expecting that the state has `external_id` defined?

Reply:
Yes. Maybe we would be safer re-reading from old API and using zone_name -> legacy_id?
Adding a check to [TPF Test Coverage Ticket](https://jira.mongodb.org/browse/CLOUDP-288855)

Reply:
In case an old sharding configuration is used the expectation from the read is that external_id is not populated. This is mainly because if we have a zone with 2 shards (e.g. 1 replication spec with num_shards = 2) there would be multiple `external_id` values associated to that replication spec.
As you mention using zone_name -> legacy_id would be the safe way to have this mapping.